qreg_gbm class

ciaran-g opened the pull request:
@jbrowell merging this into the function_names_consistency branch for now so I can update the other model fitting functions before merging into the master...

updates

- added cv_control function so it can be user across all fitting code
- default output is a list of class qreg_gbm containing the fitted models and the mqr object
- new s3 methods for summarising/printing/predicting the model class qreg_gbm
- the predict method returns a 'MultiQR' class object
- fixed internal gbm(n.core) argument to 1
- updated docs to describe kfold behaviour on gbm
- updated example and test script to test out changes
- included back compatibility for old function call

jbrowell reviewed (changes requested):
This is looking great, @ciaran-g. Love the print and summary functions. The test scripts folder is a good idea too, I'll put my gam test script here too once this is merged. I think I spotted a minor thing worth fixing in the CV set-up:

- In cv_fold, I'm not sure it is set up for other column names other than "kfold". Should it allow any string giving a column name? Line 31 seems to be the issue.

- Also, should there be a warning in the fitting function if the data includes a column called "kfold" which is not used in kfold CV? E.g. if there was a feature called "kfold" (unlikely but possible) this would get overwritten (line 80).

ciaran-g commented:
good shout @jbrowell! I've updated the functions to fix both these issues 👍 

- updated cv_control to accept column name properly...
- updated gbm script to not overwrite column in data
- returned more info in model object ---> sort_info & kfold index (if relevant)

jbrowell reviewed (approved):
Great!
